File: examples/reticulated_python/12.py
class Fruit:
	def say(self:Fruit)->str:
		return "Fruit"
class Apple:
	def say(self:Apple)->str:
		return "apple"
class Banana:
	def say(self:Banana)->str:
		return "banana"

@fields({'f':Fruit})
class Main:
	f=Fruit()
	def main(self:Main) -> str:
		self.f = Apple()
		self.f = Banana()
		return self.f.say()

print(Main().main())

File: examples/strongscript/12.ts
class Fruit { say():!string {} }
class Apple extends Fruit { say():!string { return "apple" } }
class Banana extends Fruit { say():!string { return "banana" } }

class Main {
  main():!string {
    var f:!Fruit = new Apple()
    f = new Banana()
    f.say()
  }
}

